### PR TITLE
[202205]Fix potential multi-thread issue of TSA test

### PR DESCRIPTION
**Pull request by JibinBao**
### Description of PR

Fix issue:  https://github.com/sonic-net/sonic-mgmt/issues/10119
It might be a multi-thread issue. when concurrent_tasks is 8, the case fails about 1 time per 3 times, when it is set to 4, run the test 20 times, and all pass.

Summary:
Fixes # (issue)https://github.com/sonic-net/sonic-mgmt/issues/10119

### Type of change

- [x] Bug fix
- [ ] Testbed and Framework(new/improvement)
- [ ] Test case(new/improvement)

### Back port request
- [ ] 201911
- [ ] 202012
- [ ] 202205

#### What is the motivation for this PR?
Fix issue:  https://github.com/sonic-net/sonic-mgmt/issues/10119

#### How did you do it?
decrease the multi-task number from 8  to 4

#### How did you verify/test it?
Run Test_TSA 20 times

#### Any platform specific information?
Any

### CODE REVIEW

**Review by nhe-NV — APPROVED** [review]

**Comment by wangxin** [comment]
@JibinBao Is the issue only on 202205 branch? If not, fix should go to master branch firstly. BTW, can you improve the PR title? Like "[202205] Fix potential multi-thread issue of TSA test".

**Comment by JibinBao** [comment]
> @JibinBao Is the issue only on 202205 branch? If not, fix should go to master branch firstly. BTW, can you improve the PR title? Like "[202205] Fix potential multi-thread issue of TSA test".

I just saw it happen on branch 202205. On branch 202205, we still use python2, but on master we use python3. So, maybe there is the same issue on branch master

File: tests/bgp/test_traffic_shift.py [hunk 1]
```diff
@@ -184,7 +184,7 @@ def parse_routes_process(node=None, results=None):
             routes[entry] = community
         results[hostname] = routes
 
-    all_routes = parallel_run(parse_routes_process, (), {}, neigh_hosts.values(), timeout=240, concurrent_tasks=8)
+    all_routes = parallel_run(parse_routes_process, (), {}, neigh_hosts.values(), timeout=240, concurrent_tasks=4)
```
Comment (roy-sror) [inline tests/bgp/test_traffic_shift.py:187]: Jibin, if this is a concurrency issue, why reducing the number of tasks would eliminate the problem?

Comment (JibinBao) [reply tests/bgp/test_traffic_shift.py:187]: Reducing the number of tasks may decrease the possibility of the issue. When changing it to 4 tasks,  we tested it 20 times, the issue didn't happen again. If we still use 8 tasks, this issue will happen one time about every 3 times.

**Review by bingwang-ms — APPROVED** [review]

**Comment by bingwang-ms** [comment]
MSFT ADO: 19997879

**Comment by bingwang-ms** [comment]
@yxieca Can you please help merge the PR? Thanks!

**Comment by yxieca** [comment]
@JibinBao please file an issue for the parallel execution issue. Your change can unblock test but we would like to understand why 8 threads would fail and fix it properly.

**Comment by JibinBao** [comment]
Done
https://github.com/sonic-net/sonic-mgmt/issues/10334

**Pull request merged**